vitrina/datasets/models.py

```diff
@@ -977,7 +984,61 @@ def get_metadata_objects_for_version(self):
                 )
                 meta_objects.append((metadata.pk, label))
 
+        for metadata_instance in all_metadata_instances:
+            content_type_id = metadata_instance.content_type.id
+            metadata_model = ContentType.objects.get_for_id(content_type_id).model
+            if metadata_model == "prefix":
+                label = mark_safe(
+                    f"Prefix ref: "
+                    f"<span class='tag is-success is-light is-medium prop_metadata'>{metadata_instance.name}</span>"
+                )
+
+                meta_objects.append((metadata_instance.pk, label))
+
+            if metadata_model == "paramitem":
+                paramitem_instance = ParamItem.objects.filter(metadata=metadata_instance).first()
+                param = paramitem_instance.param
+                if (
+                    param.content_type_id == ContentType.objects.filter(model="dataset").first().pk
+                    and metadata_instance.metadata_version.status == VersionStatus.DRAFT
+                ):
+                    if metadata_instance.ref:
+                        dataset_param_item_metadata_ref = metadata_instance.ref
+                    label = mark_safe(
+                        f"Param ref:  <span class='tag is-success is-light is-medium prop_metadata'>{dataset_param_item_metadata_ref}</span>"
+                        f" prepare: <span class='tag is-success is-light is-medium'>{metadata_instance.prepare}</span>"
+                    )
+                    meta_objects.append((metadata_instance.pk, label))
+
+            if metadata_model == "enumitem":
+                enumitem_instance = EnumItem.objects.filter(metadata=metadata_instance).first()
+                enum = enumitem_instance.enum
+                if (
+                    enum.content_type_id == ContentType.objects.filter(model="dataset").first().pk
+                    and metadata_instance.metadata_version.status == VersionStatus.DRAFT
+                ):
+                    if metadata_instance.ref:
+                        dataset_enum_item_metadata_ref = metadata_instance.ref
+                    label = mark_safe(
+                        f"Enum ref:  <span class='tag is-success is-light is-medium prop_metadata'>{dataset_enum_item_metadata_ref}</span> "
+                        f" prepare: <span class='tag is-success is-light is-medium'>{metadata_instance.prepare}</span>"
+                    )
+                    meta_objects.append((metadata_instance.pk, label))
+
         for model in self.model_set.all():
+            if dataset_distribution_for_model := model.distribution:
+                if dataset_distribution_metadata := dataset_distribution_for_model.metadata.first():
+                    if (
+                        dataset_distribution_metadata.pk not in dataset_distributions
+                        and dataset_distribution_metadata.metadata_version.status == VersionStatus.DRAFT
+                    ):
```

Comment:
I suggest moving these checks to separate variables (regular value assignments to the walrus ones) and avoiding the indentation:


```suggestion
            dataset_distribution_for_model = model.distribution
            dataset_distribution_metadata = dataset_distribution_for_model.metadata.first()
            is_metadata_inside_expected_distributions = dataset_distribution_metadata.pk in dataset_distributions
            is_version_draft = dataset_distribution_metadata.metadata_version.status == VersionStatus.DRAFT
            
            if dataset_distribution_for_model and dataset_distribution_metadata and (not is_metadata_inside_expected_distributions and is_version_draft):
```

It will still be multiple lines of the conditional (because of line length) but we can easily avoid the nesting introduced here.

Reply:
changed, thanks